src/shims/foreign_items.rs

```diff
@@ -202,6 +202,13 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
         if !align.is_power_of_two() {
             throw_ub_format!("creating allocation with non-power-of-two alignment {}", align);
         }
+        if let Err(AlignFromBytesError::TooLarge(_)) = Align::from_bytes(align) {
+            throw_unsup_format!(
+                "creating allocation with alignment exceeding rustc's maximum \
+                 supported value"
+            );
```

Comment:
I deliberately didn't use the exact value ``2^29``here because it would be incorrect if rustc were to change this limit value one day.